ballerina/tests/to_xml_test.bal

```diff
@@ -0,0 +1,619 @@
+                    msg = "testMapJsonToXml1 result incorrect");
+}
+
+@test:Config {
+    groups: ["toXml"]
+}
+isolated function testDefaultRecordToXml1() returns error? {
+    record {int id;} data = {id: 30};
+    xml result = check toXml(data);
+    test:assertEquals(result, xml `<id>30</id>`, msg = "testDefaultRecordToXml1 result incorrect");
+}
+
+@test:Config {
+    groups: ["toXml"]
+}
+isolated function testDefaultRecordToXml2() returns error? {
+    record {int id; string name;} data = {id: 30, name: "Asha"};
+    xml result = check toXml(data);
+    test:assertEquals(result, xml `<root><id>30</id><name>Asha</name></root>`,
+                      msg = "testDefaultRecordToXml2 result incorrect");
+}
+
+@test:Config {
+    groups: ["toXml"]
+}
+isolated function testMapStringToXml1() returns error? {
+    map<string> data = {"id": "30"};
+    xml result = check toXml(data);
+    test:assertEquals(result, xml `<id>30</id>`, msg = "testMapStringToXml1 result incorrect");
+}
+
+@test:Config {
+    groups: ["toXml"]
+}
+isolated function testMapStringToXml2() returns error? {
+    map<string> data = {"id": "30", "name": "Asha"};
+    xml result = check toXml(data);
+    test:assertEquals(result, xml `<root><id>30</id><name>Asha</name></root>`,
+                      msg = "testMapStringToXml2 result incorrect");
+}
+
+@Name {
+    value: "Customers"
+}
+@Namespace {
+    prefix: "ns",
+    uri: "http://sdf.com"
+}
+type Customer record {
+
+    @Name {
+        value: "employeeName"
+    }
+    @Attribute
+    string ns\:name;
+
+    int age;
+};
+
+@test:Config {
+    groups: ["toXml"]
+}
+isolated function testRecordWithAnnotationToXml1() returns error? {
+    Customer data = {ns\:name: "Asha", age: 10};
+    xml result = check toXml(data);
+    test:assertEquals(result,
+                    xml `<ns:Customers xmlns:ns="http://sdf.com" ns:employeeName="Asha"><age>10</age></ns:Customers>`,
+                    msg = "testRecordWithAnnotationToXml1 result incorrect");
+}
+
+@Namespace {
+    prefix: "ns",
+    uri: "http://sdf.com"
+}
+@Name {
+    value: "Customers"
+}
+type Customer2 record {
+
+    @Namespace {
+        prefix: "ns"
+    }
+    @Name {
+        value: "employeeName"
+    }
+    @Attribute
+    string name;
+
+    int age;
+};
+
+@test:Config {
+    groups: ["toXml"]
+}
+isolated function testRecordWithAnnotationToXml2() returns error? {
+    Customer2 data = {name: "Asha", age: 10};
+    xml result = check toXml(data);
+    test:assertEquals(result,
+                    xml `<ns:Customers xmlns:ns="http://sdf.com" ns:employeeName="Asha"><age>10</age></ns:Customers>`,
+                    msg = "testRecordWithAnnotationToXml2 result incorrect");
+}
+
+@Namespace {
+    prefix: "ns",
+    uri: "http://sdf.com"
+}
+@Name {
+    value: "Customers"
+}
+type Customer3 record {
+
+    @Namespace {
+        prefix: "ns"
+    }
```

Comment:
Let's remove support for adding this annotation at record field level as per above comment